core/lib/types/src/l1_batch_commit_data_generator.rs

```diff
@@ -0,0 +1,74 @@
+use zksync_basic_types::{ethabi::Token, U256};
+
+use crate::{commitment::L1BatchWithMetadata, utils};
+
+pub trait L1BatchCommitDataGenerator
+where
+    Self: std::fmt::Debug + Send + Sync,
+{
+    fn l1_commit_data(&self, l1_batch_with_metadata: &L1BatchWithMetadata) -> Token;
+}
+
+#[derive(Debug, Clone)]
+pub struct RollupModeL1BatchCommitDataGenerator {}
```

Comment:
Nit: you can just remove braces for empty structs both here and when you construct it.